### PR TITLE
Account notifications classview

### PR DESCRIPTION
**Pull request by ehfeng**
Change account notifications to classview

/cc @getsentry/team

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/3741)

### CODE REVIEW

File: src/sentry/web/frontend/account_notification.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from sentry.web.forms.accounts import ProjectEmailOptionsForm, NotificationSettingsForm
+from sentry.web.decorators import login_required
+from sentry.web.frontend.base import BaseView
+from sentry.web.helpers import render_to_response
+from sentry.utils.auth import get_auth_providers
+from sentry.utils.safe import safe_execute
+
+
+class AccountNotificationView(BaseView):
+    def get_settings_form(self, request):
```
Comment (mattrobenolt) [inline src/sentry/web/frontend/account_notification.py]: Let's just make this an attribute:

```python
class AccountNotificationView(BaseView):
    notifications_settings_form = NotificationsSettingsForm

    def handle(self, request):
        settings_form = self.notifications_settings_form(request.user, request.POST or None)
```

Then a subclass only needs to set the property.

**Comment by ehfeng** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/sentry/web/frontend/account_notification.py, line 28 \[r1\]](https://reviewable.io:443/reviews/getsentry/sentry/3741#-KMkqpX0d5dGHPJqofRZ-r1-28:-KMlAkzSYE5hb5nUI-a_:-499146009) ([raw file](https://github.com/getsentry/sentry/blob/aff1c42c14cf7e0eb4a10ee61a8ee37431f2b13f/src/sentry/web/frontend/account_notification.py#L28)):*
<details><summary><i>Previously, mattrobenolt (Matt Robenolt) wrote…</i></summary>
> Let's just make this an attribute:
> 
> 
> 
> ```python
> 
> class AccountNotificationView(BaseView):
> 
>     notifications_settings_form = NotificationsSettingsForm
> 
> 
> 
>     def handle(self, request):
> 
>         settings_form = self.notifications_settings_form(request.user, request.POST or None)
> 
> ```
> 
> 
> 
> Then a subclass only needs to set the property.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3741)*

**Comment by ehfeng** [comment]
Also, added a email_verified signal that I will be hooking into.

---

Review status: 0 of 6 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3741#-:-KMlAmasZqIXy7M9fllX:-694171984)*

**Comment by ehfeng** [comment]
Reviewed 1 of 5 files at r1.
Review status: 0 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3741)*

**Comment by ehfeng** [comment]
Reviewed 3 of 5 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3741)*

**Pull request merged**